airflow/contrib/operators/sqoop_operator.py

```diff
@@ -17,75 +17,160 @@
         self.where = where
         self.export_dir = export_dir
+        self.input_null_string = input_null_string
+        self.input_null_non_string = input_null_non_string
+        self.staging_table = staging_table
+        self.clear_staging_table = clear_staging_table
+        self.enclosed_by = enclosed_by
+        self.escaped_by = escaped_by
+        self.input_fields_terminated_by = input_fields_terminated_by
+        self.input_lines_terminated_by = input_lines_terminated_by
+        self.input_optionally_enclosed_by = input_optionally_enclosed_by
+        self.batch = batch
+        self.direct = direct
+        self.driver = driver
+        self.verbose = verbose
+        self.relaxed_isolation = relaxed_isolation
+        # No mutable types in the default parameters
+        if properties is None:
+            properties = {}
+        self.properties = properties
 
     def execute(self, context):
         """
         Execute sqoop job
         """
-        hook = SqoopHook(conn_id=self.conn_id)
+        hook = SqoopHook(conn_id=self.conn_id,
+                         verbose=self.verbose,
+                         num_mappers=self.num_mappers,
+                         properties=self.properties)
 
-        if self.type_cmd is 'export':
+        if self.cmd_type is 'export':
             hook.export_table(
                 table=self.table,
                 export_dir=self.export_dir,
-                num_mappers=self.num_mappers)
+                input_null_string=self.input_null_string,
+                input_null_non_string=self.input_null_non_string,
+                staging_table=self.staging_table,
+                clear_staging_table=self.clear_staging_table,
+                enclosed_by=self.enclosed_by,
+                escaped_by=self.escaped_by,
+                input_fields_terminated_by=self.input_fields_terminated_by,
+                input_lines_terminated_by=self.input_lines_terminated_by,
+                input_optionally_enclosed_by=self.input_optionally_enclosed_by,
+                batch=self.batch,
+                relaxed_isolation=self.relaxed_isolation)
         else:
-            hook.import_table(
-                table=self.table,
-                target_dir=self.target_dir,
-                append=self.append,
-                type=self.type,
-                columns=self.columns,
-                num_mappers=self.num_mappers,
-                split_by=self.split_by,
-                where=self.where)
+            if not self.table:
+                hook.import_table(
+                    table=self.table,
+                    target_dir=self.target_dir,
+                    append=self.append,
+                    file_type=self.file_type,
+                    columns=self.columns,
+                    split_by=self.split_by,
+                    where=self.where,
+                    direct=self.direct,
+                    driver=self.driver)
+            elif not self.query:
```

Comment:
if someone passes a query don't we want to run this function?

Reply:
The negation should be removed 👍